### ISSUE TITLE
Partition functionality in `src` missing

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#2301]
TL;DR I can haz partitions inside `src`?

Let me quote the third commandment of OSCAR development:

> Thou shalt not covet code from `experimental`

(or for heathens: "Code from `src` must never use code from `experimental`")

Now we stopped importing the partition functionality from AbstractAlgebra to export the partition functionality from `experimental/JuLie`. So we now have `Partition` etc, but we cannot use it inside `src/`. This is a rather unfortunate state. (My motivation comes from using partitions in #1986.)

<s>
I think we should do something about this, so that we can use partitions again in `Oscar/src`. I see two options
1. Move `Partition/partitions` out of `experimental/JuLie` into `src/Combinatorics`.
2. Reenable `AbstractAlgebra.Partition` as the default partition type in Oscar.

I really don't like 2. Are there any objections to 1.?
</s>

We decided to move the parts related to partitions to `src/`.

**Comment by lgoettgens** [comment]
I guess there is a third option, already in use for `Rings/binomial_ideals.jl`.

3. Include after the inclusion of experimental, like in https://github.com/oscar-system/Oscar.jl/blob/master/src/Oscar.jl#L565-L566

I do not think that this is a great idea, but I would like no mention it for further discussion.

**Comment by thofma** [comment]
How is that an option? We are not allowed to use experimental code in `src/` as far as I understand. See the documentation that I cited.

**Comment by lgoettgens** [comment]
> How is that an option? We are not allowed to use experimental code in `src/` as far as I understand. See the documentation that I cited.

Please see https://github.com/oscar-system/Oscar.jl/pull/2157#issuecomment-1488253325 from the discussion in the pr introducing this part of the documentation.

**Comment by thofma** [comment]
If I use it in `src/` it essentially becomes part of `src/` (because it now cannot be removed anymore and must be maintained) and can just be moved to `src/`. For me this is equivalent to option 1.

**Comment by thofma** [comment]
Today at the meeting, we decided that the things related to partitions should be "upgraded" to become part of `src/Combinatorics`. (Who will be doing this, is not clear to me yet.)

(Pinging @ClaudiaHeYun and @ulthiel to let them know about this change, since they have been involved in adding this to OSCAR.)
